### ISSUE TITLE
Missing migration for Django 4.0.1?

### ISSUE THREAD

**Issue by max-wittig** [jazzband/django-oauth-toolkit#1067]
**Describe the bug**

Running `makemigrations` should not have an output or create new migration files for this library.

```sh
poetry run python manage.py makemigrations --check --dry-run
Migrations for 'oauth2_provider':
  /Users/max/Library/Caches/pypoetry/virtualenvs/codeapps-wUwe-wZH-py3.10/lib/python3.10/site-packages/oauth2_provider/migrations/0005_alter_accesstoken_user_alter_application_user_and_more.py
    - Alter field user on accesstoken
    - Alter field user on application
    - Alter field user on grant
    - Alter field user on idtoken
    - Alter field user on refreshtoken
 ```

**To Reproduce**

Upgrade to Django `4.0.1` and django-oauth-toolkit `1.6.1`

**Expected behavior**

Nothing

**Version**

1.6.1

- [x] I have tested with the latest published release and it's still a problem.
- [ ] I have tested with the master branch and it's still a problem.

**Comment by n2ygk** [comment]
My understanding from @Andrew-Chen-Wang was that there was a known bug with 4.0.0 having to do with creating an unnecessary migration that was supposed to have been fixed in 4.0.1. Looks like maybe it's not fixed?

**Comment by n2ygk** [comment]
@max-wittig Can you try out https://github.com/jazzband/django-oauth-toolkit/pull/1056 and see if it works for you. It looks right to me, but I'd like some confirmation before merging it.

**Comment by max-wittig** [comment]
It's not quite fixed yet with 366e53193769e8a60e210d73c240696a4fb690c5

```sh
poetry run python manage.py makemigrations --check --dry-run
Migrations for 'oauth2_provider':
  /Users/max/Library/Caches/pypoetry/virtualenvs/codeapps-wUwe-wZH-py3.10/lib/python3.10/site-packages/oauth2_provider/migrations/0006_alter_application_client_secret.py
    - Alter field client_secret on application
```

**Comment by n2ygk** [comment]
I suspect that may be related to #1020

Lesson learned - We need to add a `makemigrations` test!

**Comment by max-wittig** [comment]
Yup. We also got this in our little Django project. Just in the CI/CD pipeline:

```sh
python manage.py makemigrations --check --dry-run
```

This will throw a 1 exit-code, when migrations are missing.

**Comment by Zorig** [comment]
@max-wittig exact same configuration on CI/CD and got the same result which is failing.

**Comment by n2ygk** [comment]
Hey @max-wittig @Zorig -- Can you please try out the stable/1.6 branch where I have a hotfix 1.6.2 release that should hopefully fix this?

**Comment by max-wittig** [comment]
Looks good to me:

```sh
poetry run python manage.py makemigrations --check --dry-run
No changes detected
```

**Comment by n2ygk** [comment]
Resolved with #1056 and published in https://pypi.org/project/django-oauth-toolkit/1.6.2/

**Comment by max-wittig** [comment]
Thank you!
